leea_agent_sdk/api.py

```diff
@@ -6,23 +8,48 @@
 
 class LeeaApi:
     def __init__(self, api_key=None):
-        self._api_host = getenv('LEEA_API_HOST', 'https://api.leealabs.com')
+        self._api_host = getenv("LEEA_API_HOST", "https://api.leealabs.com")
         self._api_key = api_key or getenv("LEEA_API_KEY")
         if not self._api_key:
             raise RuntimeError("Please provide LEEA_API_KEY")
 
         self._client = urllib3.PoolManager(
-            cert_reqs='CERT_REQUIRED',
+            cert_reqs="CERT_REQUIRED",
             ca_certs=certifi.where(),
             retries=urllib3.Retry(connect=5, read=5, redirect=5, backoff_factor=0.5),
-
         )
 
-    def _request(self, endpoint, method='GET'):
-        return self._client.request(method, f"{self._api_host}/{endpoint}", headers={"Authorization": f"Bearer {self._api_key}"}).json()
+    def _request(self, endpoint, method="GET"):
```

Comment:
I think you overcomplicating. To catch json decoding errors one try-catch is enough. 
Also, urllib3 will raise exception if status is in set `frozenset({413, 429, 503})`.

404 is fine, i'll make sure that API returns json if agent is not found